src/main/java/org/apereo/openlrs/OpenLRSAuthenticationFilter.java

```diff
@@ -111,11 +111,24 @@ private void authenticateOAuth(String authorizationHeader, HttpServletRequest re
 						normalizedParams.put(key, value);
 					}
 				}
-				
+
+				StringBuffer requestURL = request.getRequestURL();
+				String requestScheme = request.getScheme();
+				String originalScheme = request.getHeader("x-forwarded-proto");
+
+				// If original request scheme (maybe through load-balancer) is different, use it instead.
```

Comment:
Sadly, while I agree we should handle the change in port number as well, I need to put this idea on the back burner for when I have more time.

I found that I could do this:

```java
HttpRequest newRequest = new ServletServerHttpRequest(request);
UriComponentsBuilder urlBuilder = UriComponentsBuilder.fromHttpRequest(newRequest);
log.debug("URL from urlBuilder: " + urlBuilder.build(false).encode().toUriString());
```

However, while `UriComponentsBuilder` would get the request's original scheme from the `x-forwarded-proto` header, it failed to get the original port from the `x-forwarded-port` header.  Therefore, the original URL of `https://host.example.org/caliper/` would turn out to be `https://host.example.org:80/caliper/`.  If any port were included, it should be ":443", but that might cause miscalculation of the OAuth signature.

Curious about that last point, I disabled my code supporting the changed port number, and added the unnecessary port number "443" to my request URL, `https://host.example.org:443/caliper/`.  My code for handling the changed scheme worked and produced the URL, `https://host.example.org/caliper/`, without the port.  I was sure that would cause a miscalculation in the signature, but it didn't!

So, for our current configuration, dealing with the changed port number isn't necessary, as long as the code handles the changed scheme and passes the modified URL directly to `OAuthUtils.sign()`.